### DISCUSSION TITLE
Scroll bar is not at the end of screen.

### DISCUSSION THREAD

**Discussion by komiljonov** [zen-browser/desktop#2476] [Ideas]
### Captchas

- [ ] I have read the instructions.
- [ ] I have searched existing issues and avoided creating duplicates.
- [ ] I am not filing an enhancement request.

### What happened?

usually i scroll pages to down faster with mouse itself.
In some browsers i need to move mouse to end of screen to right side. and click and move mouse to scroll.

But in zen (maybe even in firefox) i need to adjust my mouse to scroll bar to be srollable.

Maybe it's not important but many users use it brainless

![image](https://github.com/user-attachments/assets/c1166999-d21b-4e11-b176-c8979552760e)

### Reproducible?

- [ ] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.15

### What platform are you seeing the problem on?

Windows

**Comment by RayZ3R0** [comment]
Not really a bug but a suggestion. Moving to discussions. Also, this helps for people who have tabs on the right like me

**Reply by komiljonov** [reply]
Yeah it' s not a bug. Suggestion. I couldn't find how to send suggestion. And sent githib issue.

**Reply by RayZ3R0** [reply]
No worries, I sent this to discussions

**Comment by tedbergstrand** [comment]
Here's my hacky fix. I think it has to do with the container that the body selector sits in. The scrollbar is at the edge of body, but the container is wider than body, so we end up with extra gap. But that's a guess based entirely on the colors.

There's also a Firefox Extension I recommend: ScrollAnywhere. It lets you set an option to use a mouse button anywhere on the page to scroll and has different "scrolling styles." I set mine up so I right click on the page and drag and it accelerates and scrolls like a smartphone. I like it a lot.

Anyways, shove this in your userChrome.css to move the scrollbar over. You may need to adjust the percentage a bit for your monitor:

```
body {
  width: 100.35% !important;
}
```

![Screenshot 2024-10-31 124239](https://github.com/user-attachments/assets/7487c558-109e-407c-936c-e02d758cb503)


Note: I'm running 1.0.1-a.12. They may have broken this fix with the changes for glance. But the general principle should still work.

**Comment by tfriedel** [comment]
I'm just starting to try out zen browser and this is the first thing I noticed that really annoyed me. Going to try the workaround, but if this is not possible it would be a  dealbreaker.

**Comment by rubaboo** [comment]
There is an "official" (from the Zen's author @mauro-balades) mod for this: [Remove Browser Padding](https://zen-browser.app/mods/680424a8-a818-406b-98c5-7726214e2a9f)
@komiljonov If you are happy with this solution, please close the discussion as resolved.

**Reply by tedbergstrand** [reply]
> There is an "official" (from the Zen's author @mauro-balades) mod for this: [Remove Browser Padding](https://zen-browser.app/mods/680424a8-a818-406b-98c5-7726214e2a9f) @komiljonov If you are happy with this solution, please close the discussion as resolved.

The Remove Browser Padding mod doesn't solve this problem. There's still too much space between body and the boundary of the monitor.  So the mouse cursor is unable to interact with the scrollbar. It's closer to the edge, but still too far away. So far, my body hack is the only way I've been able to make this work.

Was the mod you suggested able to solve the problem on your own machine?

**Reply by RayZ3R0** [reply]
Hmm if you want to almost remove that padding you can go to about:config and set `zen.theme.content-element-separation` to 1

**Reply by tedbergstrand** [reply]
> Hmm if you want to almost remove that padding you can go to about:config and set `zen.theme.content-element-separation` to 1


Also does not work. And it doesn't work at 0, either. There's still too much dead space between the scrollbar and the boundary of the monitor. I've tried *a lot* of ways to make it work and expanding body width is the only one that's been successful.

Are you trying the suggestions you're making and having success?

**Reply by rubaboo** [reply]
> Was the mod you suggested able to solve the problem on your own machine?

Yes, it was. I move the cursor all the way to the right, and I can scroll the maximized window from there. I'm on Windows. Also, I'm running 1.0.1-a.17.

**Reply by tedbergstrand** [reply]
> > Was the mod you suggested able to solve the problem on your own machine?
> 
> Yes, it was. I move the cursor all the way to the right, and I can scroll the maximized window from there. I'm on Windows. Also, I'm running 1.0.1-a.17.

Must be a case of exceptional hardware, then. Either way, it's still a problem. Neither fix suggested by @rubaboo and @RayZ3R0 work on my monitor.

**Reply by rubaboo** [reply]
> Must be a case of exceptional hardware, then. 

If you tried with 1.0.1-a.17 and it still doesn't work, perhaps it's the Window Manager, if you are using any.

**Reply by tedbergstrand** [reply]
> > Must be a case of exceptional hardware, then.
> 
> If you tried with 1.0.1-a.17 and it still doesn't work, perhaps it's the Window Manager, if you are using any.

I'm testing on standard Windows 11. I have the same result in a.17, twilight 17, as well a.12 (my preferred, least buggy version right now). 

In case it helps someone, I'm having this problem on both a Dell S3422DWG 1440p Ultrawide Monitor as well as an Acer CB272K 4K Monitor.

Here is video of each attempt:

_No Mods_ -- **Failure**
https://github.com/user-attachments/assets/355a2a70-14a3-4220-a978-13c2ec95444b

_Remove Browser Padding Mod_ -- **Failure**
https://github.com/user-attachments/assets/fdc64967-97ea-4ce4-994f-f3007305c952

_zen.theme.content-element-separation to 1 in about:config_ -- **Failure**
https://github.com/user-attachments/assets/60b308b0-1c20-4d3b-a54b-bb2584ce64b1

_Body Width Hack_ -- **Success**
https://github.com/user-attachments/assets/6ed1551f-f2fb-4461-8c56-4b5adfb8ae74

**Reply by rubaboo** [reply]
Oh, I'm on Windows 10.

**Reply by rubaboo** [reply]
Also, are you using PowerToys by any chance? Fancy Zones feature creates padding around windows.

**Reply by tedbergstrand** [reply]
> Also, are you using PowerToys by any chance? Fancy Zones feature creates padding around windows.

I thought of that, too. It happens with Fancy Zones on and off. If it's to do with zones, it's broken by the default Windows 11 zones-- which is a problem. Additionally, the problem doesn't present itself in standard Firefox, which would presumably be under the same zoning constraints.

I also don't think that would explain why expanding the body would fix it. Theoretically, the padding for the zone would still affect the window, regardless of how wide the body element is.

To me, all of this points to something funny behind the body that's causing the problem. Or maybe something to do with the scrollbar.

 Since this isn't a problem in standard Firefox, it must be something in Zen. And since we know that the Remove Browser Padding mod doesn't fix it, and neither does adjusting the element separation, there seemingly has to be another element in the UI that's gumming up the works.

I'm starting to think this should be moved back to Bugs.

**Reply by rubaboo** [reply]
Now I'm curious. Why does it work on my machine?

I have these mods installed:

![image](https://github.com/user-attachments/assets/e8adb8d8-1c11-4a30-97f9-4c65026128f2)

**Reply by tedbergstrand** [reply]
No idea. It doesn't make sense to me, either. I'm just trying to put the puzzle pieces together because I know that my body hack is really janky. I'm going to pull out another monitor and see if that makes a difference. 

But the fact that it works fine in Firefox and doesn't work in Zen indicates a problem in Zen. Even if it works on another monitor, we don't see the same behavior between FF and Zen, which means there's something that can be remedied in code.

Edit: No change with a cheapo portable 1080p monitor. So I'm having this problem on 3 different monitors at 3 different resolutions, and using 3 different versions of Zen. The only thing that fixes _any_ of them is the body width hack (though it requires different percentages for each monitor). I've also just tried changing the Scale in the Windows 11 Display Options (because I use higher scales on hi-res monitors) and that had no effect, either.

**Comment by tedbergstrand** [comment]
I've found another selector hack that "fixes" it, too. 

```
.browserSidebarContainer {
    width: 100.35% !important;
}
```

So both body and .browserSidebarContainer can be used to "fix" it by increasing their widths. 

Note that this percentage varies slightly depending on the monitor. And if you use the Remove Browser Padding mod, the percentage decreases, but still needs to be above 100%. In my case, when using Remove Browser Padding, I can drop my width to 100.1%. 100% still leaves a gap.


I've been digging through the source and other peoples' Zen mods to find selectors, but if someone can tell me how I can inspect the actual Zen Browser UI, I'm happy to spend more time poking at it to try to find the underlying cause. I haven't had much luck poking through files to find the element hierarchy, and it feels like a bad way to approach it.

I think this might be a Bug and not a Suggestion, at this point.

**Comment by Lu9-ST** [comment]
Quick addition: The Zen Mod in the store called "No Gaps" will *also* do what "Remove Browser Padding" does to the window even IN compact mode (as that is what I use)

**Reply by tedbergstrand** [reply]
> Quick addition: The Zen Mod in the store called "No Gaps" will _also_ do what "Remove Browser Padding" does to the window even IN compact mode (as that is what I use)

No Gaps fixes my issue. I don't know how I missed that one. Thanks!

**Comment by Lu9-ST** [comment]
Beta 1.0.2 b0 and b1 update: "No Gaps" mod is no longer doing the trick...

**Reply by rubaboo** [reply]
I removed most mods since I figured out how to use `userChrome.css`.
```css
/* Eliminate gaps */
:root:not([inDOMFullscreen="true"]):not([chromehidden~="location"]):not([chromehidden~="toolbar"]) {
  & #zen-tabbox-wrapper {
    margin-bottom: 0px !important;
    margin-right: 0px !important;
  }
}
```
Use with these properties in `user.js`:
```js
user_pref("zen.theme.border-radius", 0);
user_pref("zen.theme.content-element-separation", 0);
```

Mind you, once you've done all that, you may want to add this to `userChrome.css` for the split view to look good:
```css
/* Replace an insane calc that doesn't work for
   user_pref("zen.theme.content-element-separation", 0);
*/
.zen-sidebar-web-panel-splitter, .zen-split-view-splitter[orient="vertical"] {
  min-width: 6px;
}
/* Make splitter hover indicator more prominent */
.zen-sidebar-web-panel-splitter, .zen-split-view-splitter[orient="vertical"] {
  &::before {
    left: auto !important;
	height: 100% !important;
  }
}
```

**Comment by glystik** [comment]
How is there still no official and integrated fix for this? How many potential users has this turned away? This is a very central and important QoL that everyone has come to expect from browsers especially.

**Reply by Lu9-ST** [reply]
there is a fix as a mod, that was previously mentioned (and still works). it not being fixed "officially" is because *it is like that on purpose*, i don't particularly like it but clearly some people do

**Reply by glystik** [reply]
No mod has worked for me, also when removing gaps manually it even becomes worse, then mouse selects window resize edge instead of at least doing nothing like default. Haven't tried that more complex workaround with user.js, but also have my doubts, and users should not be expected to do all this extra work - this just must be fixed officially and be the default (just extend the scrollbar width to the right when in fullscreen), and for those that want it current way can just toggle this in settings.
There is already enough obstacles converting from another browser to Zen, this doesn't have to be one of them.

**Comment by tim-chron** [comment]
As of January 2026, none of the mods mentioned above ('no gaps', 'remove browser padding') are available, at least on the official "Mods" page. It is especially important on Linux since the click-and-drag scrollwheel to scroll is not available.

**Reply by mr-cheffy** [reply]
They got replaced with `zen.theme.content-element-separation` to 0 in about:config

**Reply by tim-chron** [reply]
Thank you, that is exactly what I was looking for.
